### PR TITLE
event with date

### PR DESCRIPTION
**Pull request by mariamisa**
relates #144

### CODE REVIEW

File: client/src/components/container/portal/Attendance/index.js [hunk 1]
```diff
@@ -24,35 +25,42 @@ class Attendance extends React.Component {
...
+      } = await axios.patch('/api/v1/attendance', { userCode, eventCode });
+      success(msg);
+      resetFields();
+    } catch ({
+      response: {
+        data: { msg },
+      },
+    }) {
+      error(msg);
+    }
```
Comment (AlaaTaima) [inline client/src/components/container/portal/Attendance/index.js:63]: handle error without destructuring refers to Jak point as the following : 
```suggestion
   catch (err) {
      let errorMesg;

      if (err.response) {
        errorMesg = err.response.data.msg;
      } else {
        errorMesg = 'Something went wrong, please try again later';
      }
      error(errorMesg);
    }
```

File: client/src/components/container/portal/Attendance/index.js [hunk 2]
```diff
@@ -66,29 +74,32 @@ class Attendance extends React.Component {
...
+      } = await axios.post('/api/v1/portal/logout', { pinCode: value });
+      success(msg);
+      push('/portal');
+    } catch ({
+      response: {
+        data: { msg },
+      },
+    }) {
+      error(msg);
+    }
```
Comment (AlaaTaima) [inline client/src/components/container/portal/Attendance/index.js:102]: edit error handling here as the previous comment : 
```suggestion
       catch (err) {
      let errorMesg;
      if (err.response) {
        errorMesg = err.response.data.msg;
      } else {
        errorMesg = 'Something went wrong, please try again later';
      }
      error(errorMesg);
    }
```

File: server/app.js [hunk 3]
```diff
@@ -11,6 +11,7 @@ const app = express();
 app.disabled('x-powered-by');
 
 app.use(compression());
+app.use(express.urlencoded({extended:false}))
```
Comment (AlaaTaima) [inline server/app.js:14]: **Why did you add it !!**

File: server/controller/routes/admin/viewEventsOnDate.js [hunk 4]
```diff
@@ -3,19 +3,20 @@ const moment = require('moment');
...
-      if (rows.length !== 0) {
-        const events = rows.filter(
-          (el) => moment(el.date).format('YYYY-MM-DD') === today
-        );
-        res.json(events);
+
+const viewEventsOnDate = async (req, res, next) => {
+  try{
+    const {rows}=await getEventsDate()
+      if (!rows.length) {
```
Comment (AlaaTaima) [inline server/controller/routes/admin/viewEventsOnDate.js:10]: fix the condition here : 
```suggestion
      if (rows.length ) {
```

File: client/src/components/container/portal/Attendance/index.js [hunk 5]
```diff
@@ -66,29 +74,32 @@ class Attendance extends React.Component {
...
   };
 
-  handleModalSubmit = ({
+  handleModalSubmit = async ({
     target: {
       parentNode: {
         firstChild: { value },
       },
     },
   }) => {
```
Comment (AlaaTaima) [inline client/src/components/container/portal/Attendance/index.js:83]: you should store the event   `const targetEle = e.target;`  then you can use it's properties it as the following : 
```
  handleModalSubmit = async (e) => {
    const targetEle = e.target;
    console.log(targetEle);

    const parentEle = targetEle.parentNode;
    console.log(parentEle);

    const firstEle = targetEle.parentNode.firstChild;
    console.log(firstEle);

    const value = targetEle.parentNode.firstChild.value;
    console.log(value);
    const { success, error } = this;
    try {
      const { data } = axios.post('/api/v1/portal/logout', { pinCode: value });
      success(data.msg);
      this.props.history.push('/portal');
    } catch (err) {
      error(err.message);
    }
  };
```
Imgur

File: server/app.js (hunk 3 above)
Comment (mariamisa) [reply server/app.js:14]: to test it on postman before refactor react
